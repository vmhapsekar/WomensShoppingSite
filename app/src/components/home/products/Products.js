import React from 'react'
import PropTypes from 'prop-types'
import {Grid, GRow, GCol} from '../../common/grid/Grid'
import Product from './Product'

// Pure Component which displays group of products
const Products = ({products}) => {
  return (
    <Grid>
      <GRow>
        {
          products.map((product, index) =>
            <GCol xs={12} sm={12} md={4} lg={3} key={`product-${index}`}>
              <Product image={product.productImage}
                name={product.productName}
                price={product.price}
                isExclusive={product.isExclusive}
                isSale={product.isSale} />
            </GCol>
          )
        }
      </GRow>
    </Grid>
  )
}

Products.propTypes = {
  products: PropTypes.array
}

export default Products
