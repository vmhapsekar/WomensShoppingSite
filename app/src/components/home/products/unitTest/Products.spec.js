import React from 'react'
import Products from '../Products'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'
const products = require('../../../../assets/toolkit/data/products.json')

describe('<Products />', () => {
  test('Product with Exclusive Status', () => {
    const component = shallow(<Products products={products} />)
    expect(component).toBeDefined()
    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
