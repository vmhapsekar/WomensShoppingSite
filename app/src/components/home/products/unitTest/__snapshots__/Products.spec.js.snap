// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> Product with Exclusive Status 1`] = `
<Grid>
  <GRow>
    <GCol
      key="product-0"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-1.jpg"
        isExclusive={false}
        isSale={true}
        name="Striped shirt"
        price="$18.88"
      />
    </GCol>
    <GCol
      key="product-1"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-2.jpg"
        isExclusive={false}
        isSale={false}
        name="Denim shirt"
        price="$25.44"
      />
    </GCol>
    <GCol
      key="product-2"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-3.jpg"
        isExclusive={true}
        isSale={false}
        name="Plain cotton t-shirt"
        price="$12.93"
      />
    </GCol>
    <GCol
      key="product-3"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-4.jpg"
        isExclusive={false}
        isSale={false}
        name="Plain 3/4 sleeve cotton t-shirt"
        price="$26.32"
      />
    </GCol>
    <GCol
      key="product-4"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-5.jpg"
        isExclusive={false}
        isSale={false}
        name="White dress shirt"
        price="$12.16"
      />
    </GCol>
    <GCol
      key="product-5"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-6.jpg"
        isExclusive={false}
        isSale={false}
        name="Long Sleeve Skivvy Top"
        price="$30.27"
      />
    </GCol>
    <GCol
      key="product-6"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-7.jpg"
        isExclusive={false}
        isSale={true}
        name="Puffer Vest with Hood"
        price="$24.26"
      />
    </GCol>
    <GCol
      key="product-7"
      lg={3}
      md={4}
      sm={12}
      xs={12}
    >
      <Product
        image="product-8.jpg"
        isExclusive={true}
        isSale={false}
        name="Funnel Neck Swing Top"
        price="$17.73"
      />
    </GCol>
  </GRow>
</Grid>
`;
